pkg/upgrade/update.go

```diff
@@ -210,53 +208,33 @@ func (u *simpleUpgrader) updateInPlace(ctx context.Context, cs *api.OpenShiftMan
 
 		{
 			log.Infof("deallocating %s (%s)", *vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName, *vm.InstanceID)
-			future, err := vmc.Deallocate(ctx, cs.Properties.AzProfile.ResourceGroup, "ss-"+string(role), *vm.InstanceID)
-			if err != nil {
-				return err
-			}
-
-			err = future.WaitForCompletionRef(ctx, vmc.Client)
+			_, err := vmc.Deallocate(ctx, cs.Properties.AzProfile.ResourceGroup, "ss-"+string(role), *vm.InstanceID)
 			if err != nil {
 				return err
 			}
 		}
 
 		{
 			log.Infof("updating %s", *vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName)
-			future, err := ssc.UpdateInstances(ctx, cs.Properties.AzProfile.ResourceGroup, "ss-"+string(role), compute.VirtualMachineScaleSetVMInstanceRequiredIDs{
+			_, err := ssc.UpdateInstances(ctx, cs.Properties.AzProfile.ResourceGroup, "ss-"+string(role), compute.VirtualMachineScaleSetVMInstanceRequiredIDs{
 				InstanceIds: &[]string{*vm.InstanceID},
 			})
 			if err != nil {
 				return err
 			}
-
-			err = future.WaitForCompletionRef(ctx, ssc.Client)
-			if err != nil {
-				return err
-			}
 		}
 
 		{
 			log.Infof("reimaging %s", *vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName)
-			future, err := vmc.Reimage(ctx, cs.Properties.AzProfile.ResourceGroup, "ss-"+string(role), *vm.InstanceID)
-			if err != nil {
-				return err
-			}
-
-			err = future.WaitForCompletionRef(ctx, vmc.Client)
+			_, err := vmc.Reimage(ctx, cs.Properties.AzProfile.ResourceGroup, "ss-"+string(role), *vm.InstanceID)
 			if err != nil {
 				return err
 			}
 		}
 
 		{
 			log.Infof("starting %s", *vm.VirtualMachineScaleSetVMProperties.OsProfile.ComputerName)
-			future, err := vmc.Start(ctx, cs.Properties.AzProfile.ResourceGroup, "ss-"+string(role), *vm.InstanceID)
-			if err != nil {
-				return err
-			}
-
-			err = future.WaitForCompletionRef(ctx, vmc.Client)
```

Comment:
I was chatting to @jim-minter and he seems to think the mirrored functions should have the same behaviour (be non-blocking). It will probably reduce confusion in the long run even if the code looks a bit better like this now.